camel/interpreters/db_interpreter.py

```diff
@@ -0,0 +1,172 @@
+            column_names = [desc[0] for desc in
+                            cursor.description] if cursor.description else []
+
+            conn.commit()
+            conn.close()
+
+            if not rows:
+                return "Query executed successfully. No rows returned."
+
+            # Format as text table
+            result = "\t".join(column_names) + "\n"
+            result += "\n".join(["\t".join(map(str, row)) for row in rows])
+            return result
+        except Exception as e:
+            return f"SQLite query failed: {e}"
+
+
+class PostgresConnector(DatabaseConnector):
+    r"""PostgreSQL connector for executing SQL queries.
+
+    Args:
+        host (str): PostgreSQL host address.
+        port (int): PostgreSQL port (default: 5432).
+        user (str): Username for authentication.
+        password (str): Password for authentication.
+        database (str): Name of the database to connect to.
+    """
+
+    def __init__(self, host: str, port: int, user: str, password: str,
+                 database: str):
+        self.conn_info = {
+            "host": host,
+            "port": port,
+            "user": user,
+            "password": password,
+            "dbname": database,
+        }
+
+    def run_query(self, query: str) -> str:
+        try:
+            conn = psycopg2.connect(**self.conn_info)
+            cursor = conn.cursor()
+            cursor.execute(query)
+            rows = cursor.fetchall()
+            column_names = [desc[0] for desc in
+                            cursor.description] if cursor.description else []
+
+            conn.commit()
+            conn.close()
+
+            if not rows:
+                return "Query executed successfully. No rows returned."
+
+            result = "\t".join(column_names) + "\n"
+            result += "\n".join(["\t".join(map(str, row)) for row in rows])
+            return result
+        except Exception as e:
+            return f"PostgreSQL query failed: {e}"
+
+
+class SQLQueryInterpreter(BaseInterpreter):
+    r"""Interpreter that executes SQL code blocks on a given database
+    connector.
+
+    Args:
+        connector (DatabaseConnector): An object capable of executing SQL
+        queries.
+
+    Supported code_type: "sql"
+    """
+
+    def __init__(self, connector: DatabaseConnector):
+        self.connector = connector
+
+    def run(self, code: str, code_type: str) -> str:
+        if code_type.lower() != "sql":
+            raise InterpreterError(f"Unsupported code type: {code_type}")
+        try:
+            return self.connector.run_query(code)
+        except Exception as e:
+            raise InterpreterError(f"SQL execution error: {e}")
+
+    def supported_code_types(self) -> List[str]:
+        return ["sql"]
+
+    def update_action_space(self,
+                            action_space: Dict[str, BaseToolAgent]) -> None:
+        pass  # No-op
+
+
+def DatabaseInterpreter(db_config: Dict[str, Any]) -> BaseInterpreter:
```

Comment:
Should we rename it to make it more clear?
```suggestion
def DatabaseQueryLanguageInterpreter(db_config: Dict[str, Any]) -> BaseInterpreter:
```